content/articles/automating-jobs-scheduling-with-django_cron-in-python-application/index.md

```diff
@@ -0,0 +1,394 @@
+
+    res = [int(id.strip()) for id in result] # list comprehension to strip each element of the data
+
+    return Response(res, status=status.HTTP_200_OK)
+
+
+class NewsItemView(ListAPIView):
+  permission_classes = [AllowAny]
+
+  def get_data_from_API(self):
+    """
+        This helps to return 
+        formatted data fetched from endpoint provided
+        using request.
+    """
+    result = []
+    half = 0
+    total = len(HackerNewsID.objects.all()) # getting the total ids from the db
+
+    #slicing into half based on even or odd total
+    if total % 2 == 0:
+        half = len(HackerNewsID.objects.all()) / 2
+    else:
+        half = (len(HackerNewsID.objects.all()) / 2) + 1
+
+    ids = HackerNewsID.objects.all()[:half] #slicing the queryset to get last half
+
+    # looping through the ids queryset
+    for id in ids:
+      # external endpoint for each id
+        NEWS_URL = f'https://hacker-news.firebaseio.com/v0/item/{str(id)}.json?print=pretty'
+        headers = {'user-agent': 'quickcheck/0.0.1'} 
+        response = requests.get(NEWS_URL, headers=headers)
+        data = json.loads(response.text)
+        result.append(data)
+
+    return result
+
+
+# get the latest hackernews streamed
+  def get(self, request, format=None):
+    return Response(self.get_data_from_API(),status=status.HTTP_201_CREATED)
+```
+
+
+#### Configuring routes for your endpoint
+Under the project directory, open to the `urls.py` and paste the snippet below. These are the routes for the `rest-framework`, `admin`, and the `news` app made.
+
+```python
+from django.contrib import admin
+from django.urls import path, include
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api-auth/', include('rest_framework.urls')),
+    path('api/v0/items/', include('news.urls'))
+]
+```
+
+To connect the `news` app with the `hackernewsdemo` project route, you will need to make your custom `urls.py` inside the app directory. Then add the code below and paste it in.
+
+```python
+from django.urls import path
+from .views import NewsIdView, NewsItemView
+
+urlpatterns = [
+    path('all/', NewsIdView.as_view(), name='index'),
+    # http://127.0.0.1:8080/api/v0/items/all --> lists all news id from hackernews endpoint
+    path('hackernews/', NewsItemView.as_view(), name='news-item'), 
+    # http://127.0.0.1:8080/api/v0/items/hackernews --> GET the latest hackernews from db
+]
+```
+
+#### Creating an admin site
+As a Django developer, you are familiar with the default admin page that can be configured. Executing the command below will make a brand new administration site where all activities of the database are controlled.
+
+```bash
+python manage.py createsupuer
+```
+
+Having created a superuser admin account, you will have to register the `news` app model in the `admin.py`. Add this code in the `admin.py` file.
+
+```python
+from django.contrib import admin
+from .models import HackerNewsID, QuickCheckItem, QuickCheckNews
+
```

Comment:
`... QuickCheckItem, QuickCheckNews` are not defined anywhere in your code. Please remove them.

Reply:
Done.